compiler/damlc/daml-lf-conversion/src/DA/Daml/LFConversion.hs

```diff
@@ -1660,8 +1660,34 @@ convertExternal env stdlibRef primId lfType
     | [pkgId, modStr, templName, method] <- splitOn ":" primId
     , Nothing <- lookup pkgId modStr templName =
         error $ "convertExternal: external template not found " <> primId
+    | [pkgId, modStr, templName, choiceName, method] <- splitOn ":" primId
+    , Just LF.Template {tplTypeCon,tplChoices} <- lookup pkgId modStr templName
+    , choice <- ChoiceName (T.pack choiceName)
+    , Just TemplateChoice {chcSelfBinder,chcArgBinder} <- NM.lookup choice tplChoices = do
+        let pkgRef = PRImport $ PackageId $ T.pack pkgId
+        let mod = ModuleName $ map T.pack $ splitOn "." modStr
+        let qualify tconName = Qualified { qualPackage = pkgRef , qualModule = mod , qualObject = tconName}
+        let templateDataType = TCon . qualify
+        let (choiceArg, choiceArgType) = chcArgBinder
+        case method of
+          "exercise" -> do
+            ETmLam (chcSelfBinder, TApp (TBuiltin BTContractId) (templateDataType tplTypeCon)) $
+              ETmLam (choiceArg, choiceArgType) $
+                EUpdate $ UExercise (qualify tplTypeCon) choice (EVar chcSelfBinder) Nothing (EVar choiceArg)
+          "_toAnyChoice" ->
+            -- TODO: envLfVersion env `supports` featureAnyType
```

Comment:
Can  we not add the implementation already for the lf versions that support `Any`?

Reply:
As discussed, let's leave this to a following PR